src/visualization/types.jl

```diff
@@ -348,58 +348,77 @@ end
 
 # specializes the PlotData2D constructor to return an PlotData2DTriangulated for any type of mesh.
 function PlotData2DTriangulated(u, mesh, equations, dg::DGSEM, cache;
-                                solution_variables=nothing, nvisnodes=2*polydeg(dg))
+                                solution_variables=nothing, nvisnodes=nothing,
+                                elevations=nothing, slice=:yz, point=[0,0,0])
 
-  @assert ndims(mesh) == 2 "Input must be two-dimensional."
+  # convert variables based on solution_variables mapping
+  solution_variables_ = digest_solution_variables(equations, solution_variables)
+  variable_names = SVector(varnames(solution_variables_, equations))
 
-  n_nodes_2d = nnodes(dg)^ndims(mesh)
-  n_elements = nelements(dg, cache)
+  # In the 2D-case create a 'PlotData2DTriangulated' object.
+  if ndims(mesh) == 2
+    n_nodes_2d = nnodes(dg)^ndims(mesh)
+    n_elements = nelements(dg, cache)
 
-  # build nodes on reference element (seems to be the right ordering)
-  r, s = reference_node_coordinates_2d(dg)
+    # build nodes on reference element (seems to be the right ordering)
+    r, s = reference_node_coordinates_2d(dg)
 
-  # reference plotting nodes
-  if nvisnodes == 0 || nvisnodes === nothing
-    nvisnodes = polydeg(dg) + 1
-  end
-  plotting_interp_matrix = plotting_interpolation_matrix(dg; nvisnodes=nvisnodes)
-
-  # create triangulation for plotting nodes
-  r_plot, s_plot = (x->plotting_interp_matrix*x).((r, s)) # interpolate dg nodes to plotting nodes
+    # reference plotting nodes
+    if nvisnodes == nothing
+      nvisnodes=2*polydeg(dg)
+    else nvisnodes == 0
+      nvisnodes = polydeg(dg) + 1
+    end
```

Comment:
```suggestion
    if nvisnodes === nothing
      nvisnodes = 2 * polydeg(dg)
    elseif nvisnodes == 0
      nvisnodes = polydeg(dg) + 1
    end
```
`elseif` instead of `else` with a trailing statement?